compiler/src/dotty/tools/dotc/transform/SymUtils.scala

```diff
@@ -201,6 +201,9 @@ object SymUtils {
   def rawTypeRef(using Context) =
     self.typeRef.appliedTo(self.typeParams.map(_ => TypeBounds.emptyPolyKind))
 
+  def typeRef_*(using Context) =
```

Comment:
- we don't use the `_*` convention in dotty
- there's already rawTypeRef which should do the same and not be incorrect when an argument is higher-kinded

Reply:
I tried using rawTypeRef but it fails when the type parameter is not higher kinded

Reply:
That's a bug somewhere probably 😅

Reply:
```scala
scala> enum Tag[+T]:
     |   case Int extends Tag[Int]
     | 
// defined class Tag

scala> val f: Tag[? <: AnyKind] = Tag.Int
1 |val f: Tag[? <: AnyKind] = Tag.Int
  |           ^^^^^^^^^^^^
  |           Missing type parameters for  <: AnyKind
```

Reply:
```scala
scala> enum Hoge[F[_]] {
     |   case A extends Hoge[List]
     |   case B extends Hoge[[X] =>> String]
     | }
scala> val f: Hoge[?] = Hoge.A                                                                                                                 
val f: Hoge[?[_$1]] = A
```
this does work (which is the failing test case), so ? is being represented by something else

Reply:
I think the typer takes care of using the correct bounds for wildcards. Are you constructing untyped or typed trees?

Reply:
It works if I use the totally untyped TypeBoundsTree(EmptyTree, EmptyTree)

Reply:
If you're working with untyped trees that's the correct thing to do yeah

Reply:
Sorry, I sent my last reply before yours had loaded, yeah this is done before the enum template is sent to typer